src/tox/config/__init__.py

```diff
@@ -390,7 +391,7 @@ def get(self, name, default=None):
                 return os.environ.get(name, default)
             self._lookupstack.append(name)
             try:
-                res = self.reader._replace(val)
+                res = self.reader._replace(val, unquote_path=False)
```

Comment:
How quoted paths should appear in the setenv results is an aspect I havent explored yet, and would likely build on https://github.com/tox-dev/tox/pull/1691

Despite stripping quotes as early as possible, I expect there are quite a few corner cases where the quotes are going to causes new complications, which is why I have ensured the quotes are only used when `#` is in the path, which already didnt work, so corner cases breaking isnt a regression.  Adding ` ` to the quoted path chars is less clear, as that probably worked for many simple cases, and only failed in corner cases as seen at https://github.com/tox-dev/tox/issues/924.

Reply:
Tests for setting paths in `setenv` have been added.

Spaces in paths also added.

Merging #1691 first would allow for more exploration of possible edge cases using escaping.